src/components/budget/handlers/sync-budget-records-to-funding-partners.handler.ts

```diff
@@ -100,8 +100,16 @@ export class SyncBudgetRecordsToFundingPartners
     } else {
       // TODO: mous starting at midnight on Jan 1st might cause issues with UTC vs local issues here.
       // double check how year should be calculated
-      const mouStartYear = partnership.mouStart.value.year;
-      const mouEndYear = partnership.mouEnd.value.year;
+      let mouStartYear = partnership.mouStart.value.year;
+      let mouEndYear = partnership.mouEnd.value.year;
+
+      if (!mouStartYear || !mouEndYear) {
+        mouStartYear = CalendarDate.fromISO(
+          partnership.mouStart.value.toString()
+        ).year;
+        mouEndYear = CalendarDate.fromISO(partnership.mouEnd.value.toString())
+          .year;
+      }
```

Comment:
Thanks for finding this @zkhin. This shouldn't be necessary though. It looks like the `mouStart/End.value` are strings but they TS says they are CalendarDates. TS types should always match what the actual types are, if it doesn't that's a problem.
Looking into why the ISO strings are not being transformed now.

Reply:
Yes, that's the weird part.

Reply:
![](https://tenor.com/o4u5.gif)

The transformation gets applied by the `ValidationPipe`. That's why it was working in the other branch with that pipe re-enabled in tests.